content/starter-configs/tutorials/testery.md

```diff
@@ -0,0 +1,169 @@
+
+We'll provide a [full example of this below](#example-tugboat-config-for-automated-testing-with-testery), but in the
+meantime, you'll also need to set up Testery.
+
+## Configure Testery
+
+You'll need to [create an account with Testery](https://testery.io/pricing); their free tier should be enough to get up
+and running as a proof-of-concept.
+
+At the time of this writing, Testery is configured for testing with these frameworks:
+[WebdriverIO](https://webdriver.io/), [SpecFlow](https://specflow.org/), [Cypress](https://www.cypress.io/), and
+[Nightwatch.js](https://nightwatchjs.org/).
+
+Once you've got an account, you can create a project in Testery. You'll need to select the framework you want to use for
+your tests. Then you can set up your Tugboat `config.yml` to run your Testery project's tests against the Tugboat
+environment that’s automatically generated from your pull requests.
+
+## Example Tugboat config for automated testing with Testery
+
+The Tugboat configuration is managed by a [YAML file](/setting-up-tugboat/create-a-tugboat-config-file/) at
+`.tugboat/config.yml` in the git repository. In this example Tugboat + Testery configuration, we're setting up automated
+testing for [a WordPress site](../wordpress); with comments to explain what's going on:
+
+```yaml
+services:
+  # What to call the service hosting the site.
+  php:
+    # Use PHP 7.x with Apache; this syntax pulls in the latest version of PHP 7
+    image: tugboatqa/php:7-apache
+
+    # Set this as the default service. This does a few things
+    #   1. Clones the git repository into the service container
+    #   2. Exposes port 80 to the Tugboat HTTP proxy
+    #   3. Routes requests to the preview URL to this service
+    default: true
+
+    # Wait until the mysql service is done building
+    depends: mysql
+
+    # A set of commands to run while building this service
+    commands:
+      # Commands that set up the basic preview infrastructure
+      init:
+        # Install prerequisite packages
+        - apt-get update
+        - apt-get install -y rsync
+        - apt-get install -y jq
+
+        # Turn on URL rewriting.
+        - a2enmod rewrite
+
+        # Install PHP mysqli extension
+        - docker-php-ext-install mysqli
+
+        # Install wp-cli
+        - curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar
+        - chmod +x wp-cli.phar
+        - mv wp-cli.phar /usr/local/bin/wp
+
+        # Link the docroot
+        - ln -snf "${TUGBOAT_ROOT}/docroot" "${DOCROOT}"
+
+        # Configure wordpress to work with tugboat
+        - wp --allow-root --path="${DOCROOT}" config create --dbhost=mysql --dbname=tugboat --dbuser=tugboat
+          --dbpass=tugboat --force
+
+        # Update permalinks to remove the index.php.
+        - wp --allow-root --path="${DOCROOT}" option set permalink_structure /%postname%/
+        - wp --allow-root --path="${DOCROOT}" rewrite flush --hard
```

Comment:
The applicable config gets lost in here. What do you think about removing all of the detail above and doing something like this instead:

```yaml
services:
  php:
    commands:
      init:
        ...
        # Install pip so you can get the Testery CLI.
        - apt-get update
        - apt-get install -y python3 python3-pip
      build:
        ...
        # Check for upgrades to the Testery command line interface
        - pip3 install testery --upgrade
        # Set up Testery to run tests against the Tugboat environment, using Tugboat environment variables
        - testery update-environment --create-if-not-exists --token "$TESTERY_TOKEN" --key "${TUGBOAT_PREVIEW}" --name
          "${TUGBOAT_PREVIEW}" --variable "TUGBOAT_DEFAULT_SERVICE_URL=${TUGBOAT_DEFAULT_SERVICE_URL}"
        # Start a test run.
        - testery create-test-run --token "$TESTERY_TOKEN" --git-ref "$TUGBOAT_PREVIEW_SHA" --project "testery"
          --environment "${TUGBOAT_PREVIEW}"
```